### ISSUE TITLE
Nullness checker allows invalid binary operation

### ISSUE THREAD

**Issue by AndrewShf** [eisop/checker-framework#301]
```
import org.checkerframework.checker.initialization.qual.*; 
class Demo { 
    @UnknownInitialization Integer n; 
    Demo() { 
        int y = 5 + n; 
    } 
}
```
Output: No error reported.
Expected: `assignment.type.incompatible` on line `int y = 5 + n;`

This is because in nullnessAnnotatedTypeFactory.java, it overrides `visitBinary` methods, to not do anything to check.

https://github.com/eisop/checker-framework/blob/5c0e6f9398259d76f037483f2d4ff2d5ce5d38fb/checker/src/main/java/org/checkerframework/checker/nullness/NullnessAnnotatedTypeFactory.java#L573

https://github.com/eisop/checker-framework/blob/5c0e6f9398259d76f037483f2d4ff2d5ce5d38fb/checker/src/main/java/org/checkerframework/checker/nullness/NullnessAnnotatedTypeFactory.java#L642

**Comment by wmdietl** [comment]
As expected error, I would expect an unboxing of nullable error, or something like that.
Do you think this is a separate issue from #297?

**Comment by AndrewShf** [comment]
I agree. I thought we want to post this binary operation issue as well. I'll close this issue.
